nifty-ssl/src/main/java/com/facebook/nifty/ssl/TicketSeedFileParser.java

```diff
@@ -0,0 +1,173 @@
+        this.salt = new byte[salt.length];
+        System.arraycopy(salt, 0, this.salt, 0, salt.length);
+    }
+
+    /**
+     * Creates a ticket seed file parser with the given hex-encoded salt value.
+     *
+     * @param hexSalt the hex-encoded salt value to use. Must not be null.
+     */
+    public TicketSeedFileParser(final String hexSalt) {
+        this(decodeHex(hexSalt));
+    }
+
+    /**
+     * Returns a list of tickets parsed from the ticket file. The keys are returned in a format suitable for use
+     * with netty. The first keys are the current keys, following that are the new keys and old keys.
+     *
+     * @param file the ticket seed file.
+     * @return a list of ticket keys. Current keys are first, then new keys, then old keys.
+     * @throws IOException if reading the file or parsing the JSON fails.
+     * @throws IllegalArgumentException if the JSON file does not contain any current seeds.
+     */
+    public List<SessionTicketKey> parse(final File file) throws IOException, IllegalArgumentException {
+        List<String> allSeeds = TicketSeeds.parseFromJSONFile(file).getAllSeeds();
+        return allSeeds.stream().map(this::deriveKeyFromSeed).collect(Collectors.toList());
+    }
+
+    /**
+     * Helper class used to represent the contents of a parsed ticket seed file.
+     */
+    private static class TicketSeeds {
+        final List<String> currentSeeds;
+        final List<String> newSeeds;
+        final List<String> oldSeeds;
+
+        TicketSeeds(final List<String> currentSeeds, final List<String> newSeeds, final List<String> oldSeeds) {
+            checkArgument(currentSeeds != null && currentSeeds.size() > 0, "current seeds must not be empty");
+
+            this.currentSeeds = ImmutableList.copyOf(currentSeeds);
+            this.newSeeds = ImmutableList.copyOf(newSeeds);
+            this.oldSeeds = ImmutableList.copyOf(oldSeeds);
+        }
+
+        /**
+         * Returns a list of all ticket seeds. The order is: currentSeeds, newSeeds, oldSeeds.
+         *
+         * @return a list of all ticket seeds.
+         */
+        List<String> getAllSeeds() {
+            final ImmutableList.Builder<String> builder = new ImmutableList.Builder<>();
+            return builder.addAll(currentSeeds).addAll(newSeeds).addAll(oldSeeds).build();
+        }
+
+        /**
+         * Parses the contents of the given file as JSON to construct a TicketSeeds object. See comment at the
+         * top of this file for example file format.
+         *
+         * @param file the JSON file containing the ticket seeds data.
+         * @return a TicketSeeds object.
+         * @throws IOException if reading the file or parsing the contents as JSON fails.
+         * @throws IllegalArgumentException if the file does not contain any current seeds.
+         */
+        static TicketSeeds parseFromJSONFile(final File file) throws IOException {
+            final ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Premature optimization IMO, this will not get called very frequently. Also it looks like ObjectMapper does some internal caching of serializers, whatever that means. And we have lots of example code in other java projects that instantiates a new ObjectMapper every time.